### PR TITLE
#14: Update testing to new approach

### LINKED ISSUE #14: Update testing to new approach
Fixes #13.

### PR DESCRIPTION
**Pull request by nick4fake**
Fixes #14

Updated testing.

### CODE REVIEW

File: example/main.tf [hunk 1]
```diff
@@ -14,24 +14,8 @@
...
  */
 
-locals {
-  credentials_file_path = var.credentials_file_path
-}
-
-/******************************************
-  Provider configuration
- *****************************************/
-provider "google" {
```
Comment (aaron-lane) [inline example/main.tf]: The provider should still be configured in the example, with an appropriate `version` value and no `credentials` argument.

File: main.tf [hunk 2]
```diff
@@ -24,17 +24,26 @@ resource "local_file" "cloud-datastore-index-file" {
   filename = local.path_file
 }
 
+resource "google_app_engine_application" "app" {
```
Comment (aaron-lane) [inline main.tf]: What is the motivation for adding this resource?

File: scripts/create-indexes.sh [hunk 3]
```diff
@@ -16,16 +16,12 @@
...
   exit 1
 fi
 
-CREDENTIALS=$1
-PROJECT=$2
-INDEXES_FILE=$3
+PROJECT=$1
+INDEXES_FILE=$2
 
-export CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE="$CREDENTIALS"
```
Comment (aaron-lane) [inline scripts/create-indexes.sh]: `CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE` still needs to be set for `gcloud`, similiar to how it is done in [Kubernetes Engine](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/scripts/wait-for-cluster.sh#L18-L20).

File: scripts/destroy-indexes.sh [hunk 4]
```diff
@@ -16,15 +16,12 @@
...
+if [[ $# -lt 2 ]]; then
+  echo "usage: destroy-indexes.sh <project> <indexes_file>" 1>&2
   exit 1
 fi
 
-CREDENTIALS=$1
-PROJECT=$2
-INDEXES_FILE=$3
-
-export CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE="$CREDENTIALS"
```
Comment (aaron-lane) [inline scripts/destroy-indexes.sh]: Likewise about `CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE`.

File: scripts/create-indexes.sh (hunk 3 above)
Comment (nick4fake) [reply scripts/create-indexes.sh]: What is it used for?

File: main.tf (hunk 2 above)
Comment (aaron-lane) [reply main.tf]: Okay, let's move this resource to the example and document that it is required in the README. This module should not be responsible for configuring App Engine.

File: test/setup/main.tf [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  activate_apis = [
+    "cloudresourcemanager.googleapis.com",
+    "storage-api.googleapis.com",
+    "serviceusage.googleapis.com",
+    "appengine.googleapis.com",
+    "datastore.googleapis.com",
+  ]
+}
+
+resource "google_app_engine_application" "app" {
```
Comment (aaron-lane) [inline test/setup/main.tf]: This approach looks good, but let's move it to the example as it's relevant to usage of the root module. Setup configurations should be as minimal and consistent as possible.

File: test/fixtures/main.tf [hunk 6]
```diff
@@ -14,18 +14,35 @@
...
-  source      = "../../"
-  credentials = local.credentials_file_path
-  project     = var.project
-  indexes     = file("${path.module}/yaml/index.yaml")
+  source  = "../../"
+  project = var.project_id
+  indexes = data.null_data_source.dependency.outputs.indexes
+}
+
+resource "google_app_engine_application" "app" {
```
Comment (aaron-lane) [inline test/fixtures/main.tf]: This can be removed since it's in the example now.

File: test/fixtures/variables.tf [hunk 7]
```diff
@@ -14,13 +14,13 @@
...
 
-variable "credentials_file_path" {
+variable "project_id" {
   type        = string
-  description = "Service account json auth path"
+  description = "The project to run tests against"
 }
 
-variable "project" {
+variable "location_id" {
```
Comment (aaron-lane) [inline test/fixtures/variables.tf]: This can be removed since it's in the example now.

File: test/fixtures/main.tf (hunk 6 above)
Comment (nick4fake) [reply test/fixtures/main.tf]: Not really, as fixtures don't use example

**Pull request merged**